module_util.cc

```diff
@@ -41,15 +41,15 @@ string normalized_module_name(const char* module_name) {
     if ( mod_len = strlen(module_name); mod_len >= 2 && streq(module_name + mod_len - 2, "::") )
         mod_len -= 2;
 
-    return string(module_name, mod_len);
+    return {module_name, static_cast<size_t>(mod_len)};
```

Comment:
Making `mod_len` to be of `size_t` should work and avoid the cast? `strlen()` returns `size_t`.

But... this code is about removing trailing `::` from a string. Is there a more obvious/idiomatic way these days than `strlen()`, `streq()` and pointer arithmetic ? :-)

Reply:
If only we had c++20, we'd get `td::string::starts_with` and `std::string::ends_with` as part of the standard library.